File: Php/poo/employeeSalary/library/roster.php
<?php
require_once 'library/worker.php';
require_once 'library/employee.php';

class roster{
    const MIN_WAGE = 1300000;

    public function calculateTotalSalary(employee $employee): float {
        $salary = $employee->getSalary();
        $subsidy = $this->calculateSubsidy($salary);
        $netSalary = $employee->calculateTotalSalary();
        return $subsidy + $netSalary;
    }

    public function calculateSubsidy($salary): float {
        return $salary < (2 * self::MIN_WAGE) ? self::MIN_WAGE : 0;
    }
}

File: Java/poo/employeeSalary/library/roster.java
package poo.employeeSalary.library;

public class roster {
    private static final double MIN_WAGE = 1300000;

    public double calculateSubsidy(double salary) {
        return salary < (2 * MIN_WAGE) ? MIN_WAGE : 0;
    }
    public double calculateTotalSalary(employee employee) {
        double salary = employee.getSalary();
        double subsidy = this.calculateSubsidy(salary);
        double netSalary = employee.calculateTotalSalary();
        return subsidy + netSalary;
    }
}


